docs/guides/advanced/fee_model.md

```diff
@@ -12,23 +12,134 @@ The gas fee covers the following expenses:
+
+In this model, there are 8 config options, let's walk through them:
+
+`FeeParamsV2` contains 2 dynamic prices:
+
+- `l1_gas_price` - which is used to compute the cost of submitting the proofs on L1
+- `l1_pubdata_price` - which is the cost of submitting a single byte of pubdata
+
+And config options (`FeeModelConfigV2`) contain:
+
+- `minimal_l2_gas_price` - same as above, this should cover the $ costs of running the system
+
+2 fields around the maximum capacity of the batch:
+
+- `max_gas_per_batch` - maximum amount of gas in a batch - if you set this too high, you might exceed the circuits
+  limit.
+- `max_pubdata_per_batch` - the maximum amount of pubdata that we can put in a single batch (usually it is around 128kb
+  when using calldata, and 250 when using 2 blobs)
+
+the actual cost of the batch:
+
+- `batch_overhead_l1_gas` - how much gas operator will have to pay to handle the proof on L1 (this should include
+  commitBatch, proveBatch and executeBatch costs). This should NOT include any cost that is related to pubdata.
+
+And 2 fields about who contributed to closing the batch:
+
+- `pubdata_overhead_part` - from 0 - 1
+- `compute_overhead_part` - from 0 - 1
+
+#### Cost distribution between transactions
+
+Batches are closed, when we either run out of circuits (a.k.a gas / computation) or run out of pubdata capacity (too
+much data to publish to L1 in one transaction).
+
+Closing each batch, has some cost for the operator - especially the one related to L1 costs - where operator has to
+submit a bunch of transactions, including the one to verify the proof. (these costs are counted in
+`batch_overhead_l1_gas`).
+
+Now the problem that operator is facing is, who should pay for closing of the batch. In perfect world, we'd look at the
+reason for batch closure (for example pubdata) - and then charge the transactions proportionally to the amount of
+pubdata that they used. Unfortunately this is not feasible, as we have to charge the transactions as we go, rather than
+at the end of the batch (that can have 1000s of transactions).
+
+That's why we have the logic of `pubdata_overhead_part` and `compute_overhead_part`. These represent the 'odds' whether
+pubdata or compute were the reason for the batch closure - and based on this information, we distribute the costs to
+transactions:
+
+```
+cost_of_closing_the_batch * (compute_overhead_part * TX_GAS_USED / MAX_GAS_IN_BLOCK + pubdata_overhead_part * PUBDATA_USED / MAX_PUBDATA_IN_BLOCK)
+```
+
+#### Custom base token configurations
+
+When running a system based on a custom token, all the gas values above, should refer to YOUR custom token.
+
+Example, if you're running USDC as base token, and ETH currently costs 2000$, and current L1 gas price is 30 Gwei.
+
+- `l1_gas_price` param should be set to 30 \* 2000 == 60'000 Gwei
+- `l1_pubdata_price` should be also updated accordingly (probably also multiplied by 2'000)
```

Comment:
Also, it is better to add a disclaimer that currently the bootloader does not allow the pubdata price to be higher that  1M gwei. So in case the token is too cheap, they'll have to wait until we remove this limitation

Reply:
@StanislavBreadless Why does the bootloader have this limitation?